src/org/purefn/sentenza/api.clj

```diff
@@ -195,14 +195,16 @@
   ([pipeline n args]
    (let [state  (proto/init pipeline args)
          source (channeled (proto/source pipeline state) n)
-         chs    (proto/run pipeline state source)]
+         chs    (proto/run pipeline state source)
+         completed (promise)]
      (go-loop []
        (if (nil? (<! (last chs)))
          (do
            (close! (last chs))
+           (deliver completed :done)
```

Comment:
Hm, it might be better to wait on delivering the promise until after the cleanup call.